packages/wdio-config/src/lib/ConfigParser.js

```diff
@@ -14,7 +14,8 @@ const MERGE_OPTIONS = { clone: false }
 
 export default class ConfigParser {
     constructor () {
-        this._config = DEFAULT_CONFIGS
+        this._config = DEFAULT_CONFIGS()
```

Comment:
Can we keep the configs as object and rather use the spread syntax:
```suggestion
        this._config = { ...DEFAULT_CONFIGS }
```

Reply:
for one, this will not work.
the service hooks will still hold the same reference, since the spread operator merely performs a shallow copy.

second reason I think this should be discarded from design perspective, is that it transfer the responsibility to the developers to understand and not repeat this bug again.. which proved fatal once..

Reply:
makes sense, thanks for clarifying